### PR TITLE
Upgrade 1.30.x to 1.33.x

### PR DESCRIPTION
**Pull request by aman-agrawal**
**Jira** : https://devopsmx.atlassian.net/browse/OP-22085
**Testing** : Compile success, No new TCs failed bcoz of this change, deployed to ns=cvetarget aman1603/echo:23may1 - cron trigger worked fine, echo logs are clean.
**Note** :
All commits covered - verified from sheet.
Will add/modify echo code wrt final kork & fiat versions publish (if required). This code works with fiatVersion=1-0-SNAPSHOT & korkVersion=1-0-SNAPSHOT
Will not merge it to OES-1.30.1, will create separate branch OES-1.33.x. This is done just to compare the code.

### CODE REVIEW

File: echo-core/echo-core.gradle
Comment (rahul-chekuri) [inline echo-core/echo-core.gradle:1]: pinned versions.

File: echo-api-tck/echo-api-tck.gradle
Comment (rahul-chekuri) [inline echo-api-tck/echo-api-tck.gradle:1]: pinned versions

File: build.gradle
Comment (rahul-chekuri) [inline build.gradle:1]: pinned versions

File: echo-model/echo-model.gradle
Comment (rahul-chekuri) [inline echo-model/echo-model.gradle:1]: pinned versions

File: echo-notifications/echo-notifications.gradle
Comment (rahul-chekuri) [inline echo-notifications/echo-notifications.gradle:1]: pinned versions

File: echo-notifications/echo-notifications.gradle [hunk 1]
```diff
@@ -15,30 +15,45 @@
...
+  implementation "net.logstash.logback:logstash-logback-encoder:4.11"
+  implementation 'com.fasterxml.jackson.core:jackson-core:2.12.5'
+  implementation 'com.fasterxml.jackson.core:jackson-databind:2.12.5'
+
+  implementation group: 'org.apache.groovy', name: 'groovy-json', version: '4.0.9'
+
+  implementation "jakarta.validation:jakarta.validation-api:3.0.2"
+  implementation "io.cloudevents:cloudevents-http-basic:2.5.0"
+  implementation "io.cloudevents:cloudevents-json-jackson:2.5.0"
+  implementation ("dev.cdevents:cdevents-sdk-java:0.1.2")
```
Comment (rahul-chekuri) [inline echo-notifications/echo-notifications.gradle:47]: 46 and 47 not oes 1.30.1 but from oss 1.33

File: echo-pipelinetriggers/echo-pipelinetriggers.gradle
Comment (rahul-chekuri) [inline echo-pipelinetriggers/echo-pipelinetriggers.gradle:1]: pinned versions

File: echo-artifacts/echo-artifacts.gradle
Comment (rahul-chekuri) [inline echo-artifacts/echo-artifacts.gradle:1]: pinned versions

File: echo-plugins-test/echo-plugins-test.gradle
Comment (rahul-chekuri) [inline echo-plugins-test/echo-plugins-test.gradle:1]: pinned versions

File: echo-plugins-test/echo-plugins-test.gradle [hunk 2]
```diff
@@ -3,19 +3,14 @@ apply from: "${project.rootDir}/gradle/kotlin.gradle"
...
+  implementation("net.logstash.logback:logstash-logback-encoder:7.3")
+  implementation("com.netflix.spectator:spectator-api:1.5.4")
+  implementation "org.apache.groovy:groovy:4.0.9"
   testImplementation(project(":echo-web"))
   testImplementation(project(":echo-core"))
 
   testImplementation("io.spinnaker.kork:kork-plugins")
   testImplementation("io.spinnaker.kork:kork-plugins-tck")
 
-  testRuntimeOnly("org.junit.platform:junit-platform-launcher")
```
Comment (rahul-chekuri) [inline echo-plugins-test/echo-plugins-test.gradle:13]: this dep is present in oes 1.31.1 but removed in oss 1.33

File: echo-pubsub-aws/echo-pubsub-aws.gradle
Comment (rahul-chekuri) [inline echo-pubsub-aws/echo-pubsub-aws.gradle:1]: pinned versions

File: echo-pubsub-core/echo-pubsub-core.gradle
Comment (rahul-chekuri) [inline echo-pubsub-core/echo-pubsub-core.gradle:1]: pinned versions

File: echo-pubsub-google/echo-pubsub-google.gradle
Comment (rahul-chekuri) [inline echo-pubsub-google/echo-pubsub-google.gradle:1]: pinned versions

File: echo-rest/echo-rest.gradle
Comment (rahul-chekuri) [inline echo-rest/echo-rest.gradle:1]: pinned versions

File: echo-scheduler/echo-scheduler.gradle
Comment (rahul-chekuri) [inline echo-scheduler/echo-scheduler.gradle:1]: pinned versions

File: echo-scheduler/echo-scheduler.gradle [hunk 3]
```diff
@@ -15,26 +15,36 @@
...
+  implementation "com.squareup.okhttp:okhttp-urlconnection:2.7.5"
+  implementation "com.squareup.okhttp:okhttp-apache:2.7.5"
 
   implementation "com.netflix.spectator:spectator-api"
   implementation "io.spinnaker.kork:kork-artifacts"
   implementation "io.spinnaker.kork:kork-sql"
 
   if (!rootProject.hasProperty("excludeSqlDrivers")) {
-    runtimeOnly "mysql:mysql-connector-java"
+    runtimeOnly "com.mysql:mysql-connector-j"
```
Comment (rahul-chekuri) [inline echo-scheduler/echo-scheduler.gradle:40]: we do not have` com.mysql:mysql-connector-j` in both OES 1.30.1 and OSS 1.33. We have changed this to fix MySQL issues.

File: echo-telemetry/echo-telemetry.gradle
Comment (rahul-chekuri) [inline echo-telemetry/echo-telemetry.gradle:1]: pinned versions

File: echo-test/echo-test.gradle
Comment (rahul-chekuri) [inline echo-test/echo-test.gradle:1]: pinned versions

File: echo-web/echo-web.gradle [hunk 4]
```diff
@@ -54,7 +58,7 @@ dependencies {
     }
 
     runtimeOnly "io.spinnaker.kork:kork-runtime"
-
+    implementation("com.mysql:mysql-connector-j")
```
Comment (rahul-chekuri) [inline echo-web/echo-web.gradle:61]: we do not have this `com.mysql:mysql-connector-j` in OSS 1.33

File: echo-webhooks/echo-webhooks.gradle
Comment (rahul-chekuri) [inline echo-webhooks/echo-webhooks.gradle:1]: pinned versions

File: echo-webhooks/echo-webhooks.gradle [hunk 5]
```diff
@@ -13,24 +13,33 @@
...
-  implementation "javax.validation:validation-api"
+  implementation "jakarta.validation:jakarta.validation-api:3.0.2"
+  implementation "com.netflix.spectator:spectator-ext-gc:1.5.3"
+  implementation "com.netflix.spectator:spectator-ext-jvm:1.5.3"
+  implementation "com.netflix.spectator:spectator-reg-micrometer:1.5.3"
+  implementation "net.logstash.logback:logstash-logback-encoder:4.11"
+  implementation "org.springframework.boot:spring-boot-starter-validation"
+  implementation "io.cloudevents:cloudevents-spring:2.5.0"
+  implementation "io.cloudevents:cloudevents-json-jackson:2.5.0"
+  implementation "io.cloudevents:cloudevents-http-basic:2.5.0"
```
Comment (rahul-chekuri) [inline echo-webhooks/echo-webhooks.gradle:34]: 32 to 34 are not in OES 1.30.1 but in OSS 1.33

File: settings.gradle [hunk 6]
```diff
@@ -22,8 +36,6 @@
   }
 }
 
-enableFeaturePreview("VERSION_ORDERING_V2")
```
Comment (rahul-chekuri) [inline settings.gradle:25]: the above is not removed in OES 1.30.1 but it is removed in OSS 1.33

**Review by rahul-chekuri — COMMENTED** [review]
There are some changes that are not in OES 1.30.1 but in OSS 1.33 so highlighted those as part of comments. Also, pinned versions were flagged for future cleanups.

File: echo-notifications/echo-notifications.gradle (hunk 1 above)
Comment (aman-agrawal) [reply echo-notifications/echo-notifications.gradle:47]: part of new feature : https://github.com/spinnaker/echo/pull/1295

File: echo-plugins-test/echo-plugins-test.gradle (hunk 2 above)
Comment (aman-agrawal) [reply echo-plugins-test/echo-plugins-test.gradle:13]: **afaik** :
**org.junit.platform:junit-platform-launcher** supports not only JUnit 5 tests but also tests written in other frameworks like JUnit 4, TestNG, etc.
**org.junit.jupiter:junit-jupiter-engine**  is the test engine for executing tests written using JUnit 5 (JUnit Jupiter)

but when i checked the kotlin UT within echo-plugins-test, i find that it doesnt require any of it
any suggestions ?

File: echo-scheduler/echo-scheduler.gradle (hunk 3 above)
Comment (aman-agrawal) [reply echo-scheduler/echo-scheduler.gradle:40]: right : [ref](https://devopsmx.atlassian.net/browse/OP-22234)

File: echo-web/echo-web.gradle (hunk 4 above)
Comment (aman-agrawal) [reply echo-web/echo-web.gradle:61]: https://github.com/OpsMx/echo-oes/pull/27#discussion_r1631008544

File: echo-webhooks/echo-webhooks.gradle (hunk 5 above)
Comment (aman-agrawal) [reply echo-webhooks/echo-webhooks.gradle:34]: new feature : https://github.com/spinnaker/echo/pull/1290

**Review by rahul-chekuri — APPROVED** [review]

File: echo-api-tck/echo-api-tck.gradle [hunk 7]
```diff
@@ -2,16 +2,10 @@ apply from: "${project.rootDir}/gradle/kotlin.gradle"
 
 dependencies {
   implementation(project(":echo-web"))
-
   api("org.springframework.boot:spring-boot-starter-test")
   api("dev.minutest:minutest")
+  api("com.netflix.spectator:spectator-api")
+  implementation "net.logstash.logback:logstash-logback-encoder"
 
-  testRuntimeOnly("org.junit.platform:junit-platform-launcher")
```
Comment (meerajose) [inline echo-api-tck/echo-api-tck.gradle:9]: testRuntimeOnly("org.junit.platform:junit-platform-launcher") expected to remove ?

File: echo-artifacts/echo-artifacts.gradle [hunk 8]
```diff
@@ -19,8 +19,10 @@ dependencies {
   implementation project(':echo-model')
   implementation "com.squareup.retrofit:retrofit"
   implementation "com.squareup.retrofit:converter-jackson"
-  implementation "com.jakewharton.retrofit:retrofit1-okhttp3-client:1.1.0"
```
Comment (meerajose) [inline echo-artifacts/echo-artifacts.gradle:22]: version specified for the package in 1.30.x, should go with this version right?

File: echo-core/echo-core.gradle [hunk 9]
```diff
@@ -13,22 +13,28 @@
...
     api "io.spinnaker.kork:kork-plugins"
+    api "org.apache.groovy:groovy"
+    implementation "org.apache.groovy:groovy-json"
+    implementation "net.logstash.logback:logstash-logback-encoder"
+    implementation "io.reactivex:rxjava:1.3.8"
 
+    implementation "org.apache.groovy:groovy"
     implementation "com.squareup.retrofit:retrofit"
     implementation "com.squareup.retrofit:converter-jackson"
-    implementation "com.jakewharton.retrofit:retrofit1-okhttp3-client:1.1.0"
```
Comment (meerajose) [inline echo-core/echo-core.gradle:25]: package com.jakewharton.retrofit:retrofit1-okhttp3-client version needs to taken from 1.30.x , right?

Comment (aman-agrawal) [reply echo-core/echo-core.gradle:25]: Reason to remove explicit version : [ref](https://github.com/OpsMx/kork-oes/blob/0e80faedefcb5f3e46f61d40c43149c0518c154c/spinnaker-dependencies/spinnaker-dependencies.gradle#L81)
kork-oes also pointed to same version. checked dI, 1.1.0 is coming.

File: echo-artifacts/echo-artifacts.gradle (hunk 8 above)
Comment (aman-agrawal) [reply echo-artifacts/echo-artifacts.gradle:22]: Reason to remove explicit version : [ref](https://github.com/OpsMx/kork-oes/blob/0e80faedefcb5f3e46f61d40c43149c0518c154c/spinnaker-dependencies/spinnaker-dependencies.gradle#L81)
kork-oes also pointed to same version. checked dI, 1.1.0 is coming.

File: echo-api-tck/echo-api-tck.gradle (hunk 7 above)
Comment (meerajose) [reply echo-api-tck/echo-api-tck.gradle:9]: ok Thanks

File: echo-artifacts/echo-artifacts.gradle (hunk 8 above)
Comment (meerajose) [reply echo-artifacts/echo-artifacts.gradle:22]: ok. good

File: echo-core/echo-core.gradle (hunk 9 above)
Comment (meerajose) [reply echo-core/echo-core.gradle:25]: ok. good

**Review by meerajose — APPROVED** [review]
approved

File: echo-web/src/test/java/com/netflix/spinnaker/echo/ApplicationSpec.java [hunk 10]
```diff
@@ -18,21 +18,23 @@
...
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.TestPropertySource;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-@ExtendWith(SpringExtension.class)
-@SpringBootTest(classes = {Application.class})
+/*@ExtendWith(SpringExtension.class)
```
[the remark this answers is not in the document]
Comment (aman-agrawal) [reply echo-web/src/test/java/com/netflix/spinnaker/echo/ApplicationSpec.java]: https://github.com/OpsMx/echo-oes/pull/30

**Pull request closed without merging**